src/core/manager.h

```diff
@@ -592,6 +592,10 @@ void manager_override_show_status(Manager *m, ShowStatus mode, const char *reaso
 void manager_set_first_boot(Manager *m, bool b);
 void manager_set_switching_root(Manager *m, bool switching_root);
 
+static inline double manager_get_progress(Manager *m) {
+        return MANAGER_IS_FINISHED(m) ? 1.0 : 1.0 - ((double) hashmap_size(m->jobs) / (double) m->n_installed_jobs);
```

Comment:
hmm, this sounds complex enough to do in a proper function, no need for a static inline.

also, maybe add a guard so that m->n_installed_jobs being zero is caught to avoid div-by-zero